FHEM/14_SD_WS07.pm

```diff
@@ -154,24 +147,20 @@ SD_WS07_Parse($$)
 	}
     
 	### Model specific attributes
-	if ($models{$modelkey} eq "T") {
-		addToDevAttrList($model."_".$deviceCode,"max-deviation-temp:1,2,3,4,5,6,7,8,9,10,15,20,25,30,35,40,45,50 ");
-		addToDevAttrList($model."_".$deviceCode,"offset-temp:slider,-25,1.0,25");
-	} elsif ($models{$modelkey} eq "TH") {
-		addToDevAttrList($model."_".$deviceCode,"max-deviation-temp:1,2,3,4,5,6,7,8,9,10,15,20,25,30,35,40,45,50 ");
+	if ($models{$modelkey} eq "TH") {
 		addToDevAttrList($model."_".$deviceCode,"max-deviation-hum:1,2,3,4,5,6,7,8,9,10,15,20,25,30,35,40,45,50 ");
-		addToDevAttrList($model."_".$deviceCode,"offset-temp:slider,-25,1.0,25");
-		addToDevAttrList($model."_".$deviceCode,"offset-hum:slider,-50,1.0,50");
+		addToDevAttrList($model."_".$deviceCode,"offset-hum");
 	}
     #print Dumper($modules{SD_WS07}{defptr});
     
-    my $def = $modules{SD_WS07}{defptr}{$iohash->{NAME} . "." . $deviceCode};
+    $deviceCode = $model . "_" . $deviceCode;
+    my $def = $modules{SD_WS07}{defptr}{$deviceCode};	# test for already defined devices
```

Comment:
Device oder ist hier ja bereits model_channel.

Das würde nur funktionieren, wenn wir die Definition in der Define Sub anpassen, die nur Channel als DEF haben.

Reply:
Die sub SD_WS07_Define nimmt so ziemlich alles an. Dort ist nur eine Prüfung drin, ob die Anzahl der Argumente kleiner 3 ist.
Es funktioniert also z.B. folgendes:
Das wäre die Kombination aus longId (32) und Kanal (1), wie sie auch bei autocreate verwendet wird:
define SD_WS07_TH_321 SD_WS07 SD_WS07_TH_321
Das wäre nur Kanal (1) wie sie auch bei autocreate verwendet wird:
define SD_WS07_TH_1 SD_WS07 SD_WS07_TH_1
Das funktioniert auch:
define Kellertemp SD_WS07 1
DEF ist im letzten Fall eben nur 1.

Reply:
Ja, aber ein Device in dem DEF 1 ist, das würde in der parse Routine doch nicht mehr gefunden werden.

Vielleicht fehlt mir auch die Klarheit. Meine grauen Zellen würden die letzten Tage durchgekocht. :)

Reply:
Bei autocreate funktioniert es ja richtig.
Wenn der User bei manuellem Define da etwas falsches eingibt, kann es nicht funktionieren. Da lassen sich auch nicht alle Fälle prüfen. Gibt er eine falsche Ident oder Kanal an, passt es halt nicht zum Sensor.

Reply:
Wenn ich es richtig verstanden habe, dann gibt es Definitionen da draußen im FHEM Wald bestehend aus
modell_channel  oder modell_longid

Durch das letzte Update haben wir auch
channel oder longid   
per Autocreate angelegt.

Soweit ich die Anpassungen der pase Routine verstanden habe, werden die Definitionen bestehend aus channel oder longid dort nicht mehr berücksichtigt.
Autocreate wird diese Definition aber auch nicht neu anlegen können, da der Name ja nach wie vor identisch geblieben ist.

Meine Idee war hier, dass wir in def prüfen, ob eine Definition mit model_sonstwas angelegt wird und wenn nicht, könnte man die definition on the fly anpassen, wobei ich jetzt auch nicht wüsste, woher wir das model nehmen. Hmm...

Reply:
Ich wüßte nicht, wie man das anpassen könnte. Ziel dieses PR war eigentlich, das der bestehende Bug so schnell wie möglich gefixt wird, damit nicht noch mehr User darüber stolpern. Die Definition ist jetzt wieder so wie vorher.

Reply:
Ja, ich versuche mal schnell einen Test zu schreiben. Ich bin nicht sicher, ob wir gerade einen neuen Bug erfunden haben :(